addons/account/models/account_move.py

```diff
@@ -688,7 +688,8 @@ def get_reconciliation_proposition(self, account_id, partner_id=False):
             query = """
                     SELECT a.id, b.id
                     FROM account_move_line a, account_move_line b
-                    WHERE a.amount_residual = -b.amount_residual
+                    WHERE a.ref = b.ref
+                    AND a.amount_residual = -b.amount_residual
```

Comment:
I think what qdp intended we he told you "first try to match the reference (not the residual amount at all)" was that you should remove this line and make no check on amount_residual field in this query ;)